packages/expo-ota/android/src/main/java/expo/modules/ota/OtaModule.kt

```diff
@@ -0,0 +1,84 @@
+package expo.modules.ota
+
+import android.content.Context
+import org.json.JSONObject
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.Promise
+import org.unimodules.core.interfaces.ExpoMethod
+
+class OtaModule(context: Context, private val persistence: ExpoOTAPersistence, private val updater: OtaUpdater) : ExportedModule(context) {
+
+    private var moduleRegistry: ModuleRegistry? = null
+    private var eventEmitter: UpdatesEventEmitter? = null
+
+    override fun getName(): String {
+        return NAME
+    }
+
+    override fun onCreate(moduleRegistry: ModuleRegistry) {
+        this.moduleRegistry = moduleRegistry
+        this.eventEmitter = createUpdatesEventEmitter(moduleRegistry)
+        updater.updateEvents = eventEmitter
+    }
+
+    @Suppress("unused")
+    @ExpoMethod
+    fun checkForUpdateAsync(promise: Promise) {
+        updater.downloadManifest(manifestHandler(promise)) { e -> promise.reject("E_FETCH_MANIFEST_FAILED", e) }
+    }
+
+    private fun manifestHandler(promise: Promise): (JSONObject) -> Unit = { manifest ->
+        val manifestComparator = persistence.config!!.manifestComparator
+        if (manifestComparator.shouldReplaceBundle(persistence.newestManifest, manifest)) {
+            promise.resolve(manifest.toString())
+        } else {
+            promise.resolve(false)
+        }
+    }
+
+    @Suppress("unused")
+    @ExpoMethod
+    fun reload(promise: Promise) {
+        try {
+            updater.prepareToReload()
```

Comment:
looks like reload may still need to be implemented. Is this doable?

Reply:
This is tricky.

While on Android that is easily doable and there are libraries for that (e. g. https://github.com/JakeWharton/ProcessPhoenix which I tested and it worked), on iOS we need to do it within AppDelegate.

My decision was, that we will keep it out of scope of this library, and will let users to decide which solution to use to reload their apps. Maybe we could provide them another module `expo-reload`?

This is not final decision, surely, and I will gladly discuss that. Just wanted to let you know what my view on that matter is.

Reply:
Hmmm, ok. Let's leave this as a TODO for now so it doesn't block this PR from landing.